src/source/sources/bookmark/bookmarkSource.ts

```diff
@@ -0,0 +1,119 @@
+import { workspace } from 'coc.nvim';
+import { debounce } from '../../../util';
+import { fsExists } from '../../../util';
+import { hlGroupManager } from '../../highlightManager';
+import { ExplorerSource, BaseTreeNode } from '../../source';
+import { sourceManager } from '../../sourceManager';
+import { bookmarkColumnRegistrar } from './bookmarkColumnRegistrar';
+import './load';
+import { initBookmarkActions } from './bookmarkActions';
+import { SourcePainters } from '../../sourcePainters';
+import { argOptions } from '../../../argOptions';
+import path from 'path';
+import pathLib from 'path';
+import BookmarkDB from './util/db';
+import { decode } from './util/fp';
+import { onCocBookmarkChange } from '../../../events';
+
+export interface BookmarkNode extends BaseTreeNode<BookmarkNode, 'root' | 'child'> {
+  fullpath: string;
+  filename: string;
+  lnum: number;
+  line: string;
+  annotation: string | undefined;
+}
+
+interface BookmarkItem {
+  line: string;
+  filetype: string;
+  annotation?: string;
+}
+
+const hl = hlGroupManager.linkGroup.bind(hlGroupManager);
+
+export const bookmarkHighlights = {
+  title: hl('BookmarkRoot', 'Constant'),
+  expandIcon: hl('BookmarkExpandIcon', 'Special'),
+  filename: hl('BookmarkFilename', 'String'),
+  fullpath: hl('BookmarkFullpath', 'Directory'),
+  position: hl('BookmarkLinenum', 'Special'),
```

Comment:
I think I should use BookmarkPosition

```suggestion
  position: hl('BookmarkPosition', 'Special'),
```